x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/alerts_flyout/alerts_flyout.tsx

```diff
@@ -56,26 +56,91 @@ export const AlertsFlyout: React.FunctionComponent<AlertsFlyoutProps> = ({
   let Body: AlertTableFlyoutComponent;
   let Footer: AlertTableFlyoutComponent;
 
+  const {
+    header: internalHeader,
+    body: internalBody,
+    footer: internalFooter,
+  } = alertsTableConfiguration?.useInternalFlyout != null
+    ? alertsTableConfiguration?.useInternalFlyout()
+    : {
+        header: null,
+        body: null,
+        footer: null,
+      };
+
   switch (state) {
     case AlertsTableFlyoutState.external:
       Header = alertsTableConfiguration?.externalFlyout?.header ?? AlertsFlyoutHeader;
       Body = alertsTableConfiguration?.externalFlyout?.body ?? null;
       Footer = alertsTableConfiguration?.externalFlyout?.footer ?? null;
       break;
     case AlertsTableFlyoutState.internal:
-      Header = alertsTableConfiguration?.internalFlyout?.header ?? AlertsFlyoutHeader;
-      Body = alertsTableConfiguration?.internalFlyout?.body ?? null;
-      Footer = alertsTableConfiguration?.internalFlyout?.footer ?? null;
+      Header = internalHeader ?? AlertsFlyoutHeader;
+      Body = internalBody;
+      Footer = internalFooter;
       break;
   }
 
-  const passedProps = {
-    alert,
-    isLoading,
-  };
+  const passedProps = useMemo(
+    () => ({
+      alert,
+      isLoading,
+    }),
+    [alert, isLoading]
+  );
+
+  const FlyoutBody = useCallback(
+    () =>
+      Body ? (
+        <Suspense fallback={null}>
+          <Body {...passedProps} />
+        </Suspense>
+      ) : null,
+    [Body, passedProps]
+  );
+
+  const FlyoutFooter = useCallback(
+    () =>
+      Footer ? (
+        <Suspense fallback={null}>
+          <Footer {...passedProps} />
+        </Suspense>
+      ) : null,
+    [Footer, passedProps]
+  );
+
+  const getFlyoutBody = useCallback(() => {
+    if (FlyoutBody) {
```

Comment:
nitpick: I think `FlyoutBody` should always exist since its a function, did you mean to use `useMemo` for FlyoutBody?

Reply:
tell me if I am wrong here but it need to be a function like `() => <></>` to works, useMemo won't do it to render a component here